### PR TITLE
RUM-4507 Add OpenTelemetry use case into the Wear sample app

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: sample/wear/proguard-rules.pro [hunk 1]
```diff
@@ -18,4 +18,7 @@
 
 # If you keep the line number information, uncomment this to
 # hide the original source file name.
-#-renamesourcefileattribute SourceFile
\ No newline at end of file
+#-renamesourcefileattribute SourceFile
+
+# These annotations are introduced by OpenTelemetry API and we need to make sure R8 will not complain about them
+-dontwarn  com.google.**
```
Comment (0xnm) [inline sample/wear/proguard-rules.pro:24]:
```suggestion
-dontwarn com.google.**
```

File: sample/kotlin/proguard-rules.pro [hunk 2]
```diff
@@ -25,3 +25,5 @@
 -dontwarn org.slf4j.impl.StaticLoggerBinder
 -dontwarn org.slf4j.impl.StaticMDCBinder
 -dontwarn reactor.blockhound.integration.BlockHoundIntegration
+# These annotations are introduced by OpenTelemetry API and we need to make sure R8 will not complain about them
+-dontwarn  com.google.**
```
Comment (0xnm) [inline sample/kotlin/proguard-rules.pro:29]:
```suggestion
-dontwarn com.google.**
```

File: sample/wear/build.gradle.kts [hunk 3]
```diff
@@ -32,6 +32,7 @@ android {
     namespace = "com.datadog.android.wear.sample"
 
     compileOptions {
+        isCoreLibraryDesugaringEnabled = true
```
Comment (0xnm) [inline sample/wear/build.gradle.kts:35]: so the only channel now to check how this Otel module behaves without desugaring enabled on the pre-24 APIs is now oncall app?

**Review by 0xnm — APPROVED** [review]

File: sample/wear/build.gradle.kts (hunk 3 above)
Comment (mariusc83) [reply sample/wear/build.gradle.kts:35]: nope also in Shopist...we have the Shopist running on 3 different device models

**Pull request merged**